### DISCUSSION TITLE
How to emit multiple files from one source

### DISCUSSION THREAD

**Discussion by vs-cristian** [webpack/webpack#16494] [Q&A]
Hi, does webpack plugin/loader API allow you when importing a file to emit several files and to connect with the loader so it will return a custom object with all emitted file's paths?

For example to generate several images from one:
```
import image from 'image.png'

// image: { 
//  1x: /some-hash/image-1x.png,
//  2x: /some-hash/image-2x.png,
//  3x: /some-hash/image-3x.png,
// }
```

**Comment by alexander-akait** [comment]
You can create own loader for this, anyway I reccomend to look at https://github.com/webpack-contrib/image-minimizer-webpack-plugin, I think you can solve your problem using it
